### PR TITLE
refactoring shipment api call

### PR DESCRIPTION
**Pull request by schmitzcarsten**
add shopware saas fixes

### CODE REVIEW

**Review by barbieswimcrew — APPROVED** [review]

File: src/Facade/MollieShipment.php [hunk 1]
```diff
@@ -70,8 +70,9 @@ public function setShipment(string $orderDeliveryId, Context $context): bool
         }
 
         $customFields = $order->getCustomFields();
+        $mollieOrderId = $customFields[CustomFieldsInterface::MOLLIE_KEY][CustomFieldsInterface::ORDER_KEY] ?? null;
```
Comment (boxblinkracer) [inline src/Facade/MollieShipment.php:73]: i think that might do a warning without isset, i would recommend adding that one again

File: src/Service/MolliePaymentExtractor.php [hunk 2]
```diff
@@ -25,38 +27,31 @@ public function extractLast(?OrderTransactionCollection $collection): ?OrderTran
...
         // sort all transactions chronological
-        $collection->sort(function (OrderTransactionEntity $a, OrderTransactionEntity $b) {
+        $collection->sort(static function (OrderTransactionEntity $a, OrderTransactionEntity $b) {
             return $a->getCreatedAt() > $b->getCreatedAt();
         });
 
         $lastTransaction = $collection->last();
 
-        if ($this->isMolliePayment($lastTransaction)) {
+        if ($lastTransaction instanceof OrderTransactionEntity && $this->isMolliePayment($lastTransaction)) {
```
Comment (boxblinkracer) [inline src/Service/MolliePaymentExtractor.php:41]: if there are mulitiple conditions, maybe early return?

**Review by boxblinkracer — APPROVED** [review]
looks good in general, please see commens, otherwise approved

File: src/Facade/MollieShipment.php (hunk 1 above)
Comment (schmitzcarsten) [reply src/Facade/MollieShipment.php:73]: no, this will not create a warning. It's like isset

**Pull request merged**